File: java/Herencia/CuentasBancarias/src/cuentasBancarias/CuentaCorriente.java
package cuentasBancarias;

public class CuentaCorriente extends Cuenta {

    private double descubierto;

    public CuentaCorriente(int dni, double descubierto) {
        super(dni);
        this.descubierto = descubierto;
    }

    @Override
    public boolean hayDineroSuficiente(double monto) {
        return super.getSaldo() + this.descubierto >= monto;
    }

    @Override
    public double get_dinero_disponible() {
        return super.getSaldo() + this.descubierto;
    }

    @Override
    public String toString() {
        return super.toString() + ", descubierto: " + this.descubierto;
    }

    public static void main(String[] args) {

        Cuenta cc = new CuentaCorriente(12345678, 50000);
        cc.depositar(30000);
        cc.extraer(60000);
        cc.extraer(20000);
        System.out.println(cc);
        System.out.println(cc.get_dinero_disponible());

    }

}

File: python/herencia/cuentas_bancarias/cuenta_corriente.py
"""Class Cuenta Corriente"""

from cuentas_bancarias.cuenta import Cuenta


class CuentaCorriente(Cuenta):
    """Class Cuenta Corriente
    Las cuentas corrientes pueden girar en descubierto hasta  un monto
    que se define al momento de su creación.
    """

    def __init__(self, dni, descubierto):
        super().__init__(dni)
        self.__descubierto = descubierto

    def get_descubierto(self):
        """Devuelve el descubierto definido en la cuenta"""
        return self.__descubierto

    def hay_dinero_suficiente(self, monto) -> bool:
        return super().get_saldo() + self.__descubierto >= monto

    def get_dinero_disponible(self) -> float:
        return super().get_saldo() + self.get_descubierto()

    def __str__(self) -> str:
        return super().__str__() + ", Descubierto: " + str(self.__descubierto)


if __name__ == "__main__":

    cc = CuentaCorriente(12345678, 10000)
    print(cc)
    cc.depositar(10000)
    print(cc)
    cc.extraer(15000)
    print(cc)
